p2p/protocol/identify/id.go

```diff
@@ -47,7 +67,8 @@ const transientTTL = 10 * time.Second
 //  * Our IPFS Agent Version
 //  * Our public Listen Addresses
 type IDService struct {
-	Host host.Host
+	Host      host.Host
+	UserAgent string
```

Comment:
Keeping this exported could lead to races if the user modifies the value at runtime. Not a biggie, but just pointing it out.